src/Take.Blip.Builder.UnitTests/Actions/ProcessHttpActionTests.cs

```diff
@@ -74,6 +76,48 @@ public async Task ProcessPostActionShouldSucceed()
             await Context.Received(1).SetVariableAsync(settings.ResponseBodyVariable, "Some result", Arg.Any<CancellationToken>());
         }
 
+        [Theory]
+        [InlineData("http://169.254.169.254/")]
+        [InlineData("http://[fd00:ec2::254]/")]
+        [InlineData("http://168.63.129.16/")]
+        [InlineData("http://10.0.0.0/")]
+        [InlineData("http://172.16.0.0/")]
+        [InlineData("http://192.168.0.0/")]
+        public async Task ProcessActionWithDeniedUriShouldFailed(string uri)
+        {
+            var settings = new ProcessHttpSettings
+            {
+                Uri = new Uri(uri),
+                Method = HttpMethod.Post.ToString(),
+                ResponseBodyVariable = "httpResultBody",
+                ResponseStatusVariable = "httpResultStatus",
+
+            };
+
+            var target = GetTarget();
+
+            var httpResponseMessage = new HttpResponseMessage()
+            {
+                StatusCode = HttpStatusCode.Accepted,
+                Content = new StringContent("Some result")
+            };
+
+            HttpClient.SendAsync(Arg.Any<HttpRequestMessage>(), Arg.Any<CancellationToken>())
+                .Returns(httpResponseMessage);
+
+            // Act
+            try
+            {
+                await target.ExecuteAsync(Context, JObject.FromObject(settings), CancellationToken);
+                throw new Exception();
+            }
+            catch (ValidationException exception)
+            {
+                // Assert
+                exception.Message.ShouldBe($"The '{uri}' settings value is invalid.");
+            }
```

Comment:
I suggest replacing these lines with the following code:
```
var exception = await target.ExecuteAsync(Context, JObject.FromObject(settings), CancellationToken).ShouldThrowAsync<ValidationException>();
exception.Message.ShouldContain($"The '{uri}' settings value is invalid.");
```

Reply:
Done!